[cmocka] Initial port for cmocka

ansasaki opened the pull request:
This adds an initial port for cmocka.

cmocka is "an elegant unit testing framework for C with support for mock objects. It only requires the standard C library, works on a range of computing platforms (including embedded) and with different compilers."

Home: https://cmocka.org/
Git: git://git.cryptomilk.org/projects/cmocka.git
Releases: https://cmocka.org/files/

Fixes #4728

ansasaki committed:
[cmocka] Initial port for cmocka

Signed-off-by: Anderson Toshiyuki Sasaki <ansasaki@redhat.com>

msftclas commented:
[![CLA assistant check](https://cla.opensource.microsoft.com/pull/badge/not_signed)](https://cla.opensource.microsoft.com/Microsoft/vcpkg?pullRequest=4790) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.opensource.microsoft.com/Microsoft/vcpkg?pullRequest=4790) before we can accept your contribution.<br/><br/>:x: ansasaki [sign now](https://cla.opensource.microsoft.com/Microsoft/vcpkg?pullRequest=4790)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.opensource.microsoft.com/check/Microsoft/vcpkg?pullRequest=4790) it.</sub>

Codiferous committed:
Only build cmocka as a static library

Cmocka won't work in Windows without copying a DLL to the appropriate
place. Building as a static library removes the need for a DLL.

vcpkg only supports static linking on Linux/macOS anyway, so thischange
unifies the platforms.

Codiferous commented:
@ansasaki Sorry for the delay in processing this PR -- the changes look good. The only thing preventing me from merging this PR is that you have no yet signed the Contributor License Agreement. We're unable to accept any PRs from contributors who have not signed the agreement. Could you take a look?

Codiferous commented:
I'm going to close this PR because we're trying to close outstanding PRs (so that we're better able to respond to incoming PRs and those that have languished). I'm happy to take a look at this as soon as the CLA is signed -- just open another PR! 😄
